server/gokbg3/grails-app/services/com/k_int/ESSearchService.groovy

```diff
@@ -318,33 +322,29 @@ class ESSearchService{
     }
   }
 
-  private void addStatusQuery(query, errors, status) {
-    if (!status){
-      query.must(QueryBuilders.termQuery('status', 'Current'))
-      return
-    }
-    QueryBuilder statusQuery = QueryBuilders.boolQuery()
-    if (status.getClass().isArray() || status instanceof List){
-      status.each {
-        addStatusToQuery(it, statusQuery)
+  private void addRefdataQuery(query, errors, field, value) {
+    QueryBuilder refdataQuery = QueryBuilders.boolQuery()
+    if (value.getClass().isArray() || value instanceof List){
+      value.each {
+        addRefdataToQuery(it, refdataQuery, field)
       }
     }
-    if (status instanceof String){
-      addStatusToQuery(status, statusQuery)
+    if (value instanceof String){
+      addRefdataToQuery(value, refdataQuery, field)
     }
-    statusQuery.minimumNumberShouldMatch(1)
-    query.must(statusQuery)
+    refdataQuery.minimumNumberShouldMatch(1)
+    query.must(refdataQuery)
     return
   }
 
 
-  private void addStatusToQuery(String status, QueryBuilder statusQuery){
+  private void addRefdataToQuery(String value, QueryBuilder refdataQuery, String field){
     try{
-      status = RefdataValue.get(Long.valueOf(status))
+      value = RefdataValue.get(Long.valueOf(value))?.value
     }
     catch (Exception e){
     }
-    statusQuery.should(QueryBuilders.termQuery('status', status))
+    refdataQuery.should(QueryBuilders.matchQuery(field, value))
   }
```

Comment:
this is quite a wild construct - who invented it?

Reply:
Do you mean the whole method or the line?